src/Map/Map.js

```diff
@@ -114,20 +115,27 @@ class Map extends React.Component {
   initializeSelect = map => {
     const { selectInteraction } = this.props
 
+    // check the map to see if select interaction has been added
+    let selectInteractionOnMap = map.getInteractions().getArray()
+      // Layer panel also adds a select interaction
+      .filter(interaction => interaction._ol_kit_interaction_type !== '_ol_kit_layer_panel_hover')
+      // this checks if the select interaction created or passed in is the same instance on the map and never double adds
+      .find(interaction => interaction instanceof olInteractionSelect)
+
     if (selectInteraction) {
       // if select is passed as a prop always use that one first
       this.selectInteraction = selectInteraction
+    } else if (selectInteractionOnMap) {
+      this.selectInteraction = selectInteractionOnMap
```

Comment:
add ugh warn to inform IA's that we are assigning internal select to this select interaction on the map

Reply:
remove the else if and add a nice ugh warn message